### ISSUE TITLE
Tree crashes when trying to set `expandedKeys = "all"`

### ISSUE THREAD

**Issue by larrifax** [adobe/react-spectrum#6568]
### Provide a general summary of the issue here

I'm trying to use the new Tree component from `react-aria-components`, but it crashes when attempting to set `expandedKeys` prop to `"all"` (which should be a valid value according to the types).

See reproduction below

### 🤔 Expected Behavior?

Tree should successfully expand all possible keys

### 😯 Current Behavior

Tree crashes with error:

```
expandedKeys.has is not a function
visitNode
https://lz5k72.csb.app/node_modules/react-aria-components/dist/Tree.main.js:418:71
visitNode
https://lz5k72.csb.app/node_modules/react-aria-components/dist/Tree.main.js:421:60
$c6576bb58bfa084b$var$flattenTree
https://lz5k72.csb.app/node_modules/react-aria-components/dist/Tree.main.js:423:33
new $c6576bb58bfa084b$var$TreeCollection
https://lz5k72.csb.app/node_modules/react-aria-components/dist/Tree.main.js:103:64
```

### 🖥️ Steps to Reproduce

https://codesandbox.io/p/sandbox/priceless-rgb-lz5k72

1. Click the `Expand explicit` button. Expansion is performed as expected
2. Click the `Expand all` button. Tree crashes and error message is displayed

### Version

1.2.1

### What browsers are you seeing the problem on?

Chrome

### What operating system are you using?

Windows

**Comment by larrifax** [comment]
Looks like I've might have jumped the gun a bit here wrt. this being a bug vs a feature request (?).

The types doesn't really allow `"all"` in its type, but Typescript doesn't complain either when you supply it. 🤷 
Also, the `convertExpanded()`-function [that's used on the incoming `expandedKeys`-prop](https://github.com/adobe/react-spectrum/blob/278e5167ffc85569c00ac32a17c595ba4c861c63/packages/react-aria-components/src/Tree.tsx#L166-L170) seems to support `"all"` as a value, so it tricked me into believing that it was intended to work.

**Comment by snowystinger** [comment]
We had a talk about this recently. We've decided that expandedKeys = 'all' isn't something we want to support. The reason being that we couldn't think of a reason why the server would need to know something like this, and selectedKeys="all" exists because all the items on the client may not be ALL the items on the server.

expanded is just the client view, so we think it's ok to gather all your keys into a Set and use that. When new items arrive from the server, you can just add those keys to what you already have.

In addition, we've gotten a good amount of feedback that selectedKeys = "all" is not very intuitive, so in the future if there was a reason to implement a feature like this, I think we'd reconsider the choice we made there.

Thanks for the issue! I'm going to close it out as not planned.
